crates/turborepo-lib/src/run/mod.rs

```diff
@@ -75,31 +79,7 @@ impl Run {
             .validate()
             .context("Invalid package dependency graph")?;
 
-        let g = CompleteGraph::new(&pkg_dep_graph, &self.base.repo_root);
-
-        let is_single_package = opts.run_opts.single_package;
-        let turbo_json = g.get_turbo_config_from_workspace(ROOT_PKG_NAME, is_single_package)?;
-
-        opts.cache_opts.remote_cache_opts = turbo_json.remote_cache_opts.clone();
-
-        if opts.run_opts.experimental_space_id.is_none() {
-            opts.run_opts.experimental_space_id = turbo_json.space_id.clone();
-        }
-
-        let pipeline = &turbo_json.pipeline;
-
-        let mut filtered_pkgs =
-            scope::resolve_packages(&opts.scope_opts, &self.base, &pkg_dep_graph)?;
-
-        if filtered_pkgs.len() == pkg_dep_graph.len() {
-            for target in targets {
-                let key = task_id::root_task_id(target);
-                if pipeline.contains_key(&key) {
-                    filtered_pkgs.insert(task_id::ROOT_PKG_NAME.to_string());
-                    break;
-                }
-            }
-        }
+        let _filtered_pkgs = scope::resolve_packages(&opts.scope_opts, &self.base, &pkg_dep_graph)?;
```

Comment:
This is getting deleted mostly because the lifetimes weren't worth resolving here and the filter/scope code will be done in a different PR.